src/js/actions/documents.js

```diff
@@ -1099,7 +1099,7 @@ define(function (require, exports) {
 
         return this.transfer(initActiveDocument);
     };
-    beforeStartup.reads = [locks.PS_DOC];
+    beforeStartup.reads = [locks.PS_DOC, locks.JS_APP];
```

Comment:
So, maybe just a comment would be better here? If we just leave this superfluous lock, it will be confusing later.